engine/l_studio.cpp

```diff
@@ -906,7 +908,7 @@ class CModelRender : public IVModelRender,
 	void DrawModelExecute( IMatRenderContext *pRenderContext, const DrawModelState_t &state, const ModelRenderInfo_t &pInfo, matrix3x4_t *pCustomBoneToWorld = NULL );
 
 	void InitColormeshParams( ModelInstance_t &instance, studiohwdata_t *pStudioHWData, colormeshparams_t *pColorMeshParams );
-	CColorMeshData *FindOrCreateStaticPropColorData( ModelInstanceHandle_t handle );
+    std::shared_ptr<CColorMeshData> FindOrCreateStaticPropColorData( ModelInstanceHandle_t handle );
```

Comment:
I would advise against using std:: stuff in the engine, you should instead use one of the custom classes  in public/tierX (maybe CRefPtr), that way someone later can go and optimize that class and have it affect the entire codebase

Reply:
I found no ready-to-use primitives there. CRefPtr and other stuff will require CColorMeshData to implement ref counting inside it. It also have to be thread safe. I'd better rely it on using std::shared_ptr instead. First, correctness is better then speed. And second, I don't think it will impact performance here - there's plenty of code between creation and release.
I would insist on accepting it as is and improving in another PR.